src/main/java/com/todoist/pojo/Collaborator.kt

```diff
@@ -0,0 +1,79 @@
+package com.todoist.pojo
+
+open class Collaborator @JvmOverloads constructor(
+        id: Long,
+        email: String,
+        fullName: String = "",
+        imageId: String? = null,
+        projectsActive: Collection<Long>? = null,
+        projectsInvited: Collection<Long>? = null,
+        deleted: Boolean = false
+) : Person(id, email, fullName, imageId, deleted) {
+
+    /**
+     * @return Unmodifiable set of active project ids.
+     */
+    var projectsActive: Set<Long> = projectsActive.orEmpty().toSet()
+        private set
+    /**
+     * @return Unmodifiable set of invited project ids.
+     */
+    var projectsInvited: Set<Long> = projectsInvited.orEmpty().toSet()
+        private set
+
+    fun getProjectState(projectId: Long): String {
+        return when {
+            projectsActive.contains(projectId) -> STATE_ACTIVE
+            projectsInvited.contains(projectId) -> STATE_INVITED
+            else -> STATE_DELETED
+        }
+    }
+
+    /**
+     * Sets the state for a specific project id.
+     */
+    open fun setProjectState(projectId: Long, state: String) {
+        when (state) {
+            STATE_ACTIVE -> {
+                projectsActive = projectsActive + projectId
+                projectsInvited = projectsInvited - projectId
+            }
+
+            STATE_INVITED -> {
+                projectsActive = projectsActive - projectId
+                projectsInvited = projectsInvited + projectId
+            }
+
+            STATE_DELETED -> {
+                projectsActive = projectsActive - projectId
+                projectsInvited = projectsInvited - projectId
+            }
+
+            else -> throw IllegalArgumentException("Unknown state.")
+        }
+    }
+
+    /**
+     * Copies the active project ids into an unmodifiable set.
+     *
+     * @see {@link .setProjectState
+     */
+    fun setProjectsActive(projectsActive: Collection<Long>) {
```

Comment:
Because the `setter` method allows for setting any `collection` whereas if we had the property setter as public, we could only set `Set`s, as we can't customize the property setter parameter type. Do you see a better way of doing it ? 🤔

Reply:
But is it useful to allow any collection? I.e. for `Int` properties we don't allow `Number` as arg and then convert them.

Reply:
This was the way it was before, e.g.: https://github.com/Doist/TodoistPojos/blob/master/src/main/java/com/todoist/pojo/Collaborator.java#L76-L78

So I just kept the workflow. I'd say we can keep it for now ? To don't break any existing workflow and then we can re-evaluate ?

Reply:
Okay, let's do that. :)